addons/point_of_sale/models/pos_category.py

```diff
@@ -30,7 +30,7 @@ def _check_category_recursion(self):
     def _get_hierarchy(self) -> List[str]:
         """ Returns a list representing the hierarchy of the categories. """
         self.ensure_one()
-        return (self.parent_id._get_hierarchy() if self.parent_id else []) + [self.name]
+        return (self.parent_id._get_hierarchy() if self.parent_id else []) + ([self.name] if self.name else [])
```

Comment:
Maybe we should use a placeholder instead? For example `Category 1 / Category 2 / New`
```suggestion
        return (self.parent_id._get_hierarchy() if self.parent_id else []) + [self.name or _('New')]```